ingestion/src/metadata/utils/credentials.py

```diff
@@ -137,3 +145,39 @@ def generate_http_basic_token(username, password):
     """
     token = base64.b64encode(f"{username}:{password}".encode("utf-8")).decode("utf-8")
     return token
+
+
+def get_default_credentials(quota_project_id: Optional[str] = None,
```

Comment:
nit - It might be helpful to set the function name with `gcp` or `google`, as the credentials module is not tied to GCP. Something like `get_gcp_default_credentials`. Same with `get_gcp_impersonate_credentials`. Thanks

Reply:
I got it. I will rename the function so that it gets more intuitive.